File: external/google-tv-pairing-protocol/cpp/src/polo/pairing/message/pairingrequestmessage.cc
// Copyright 2012 Google Inc. All Rights Reserved.
//
// Licensed under the Apache License, Version 2.0 (the "License");
// you may not use this file except in compliance with the License.
// You may obtain a copy of the License at
//
//     http://www.apache.org/licenses/LICENSE-2.0
//
// Unless required by applicable law or agreed to in writing, software
// distributed under the License is distributed on an "AS IS" BASIS,
// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
// See the License for the specific language governing permissions and
// limitations under the License.

#include "polo/pairing/message/pairingrequestmessage.h"

#include <sstream>
#include <string>

namespace polo {
namespace pairing {
namespace message {

PairingRequestMessage::PairingRequestMessage(
    const std::string& service_name)
    : PoloMessage(PoloMessage::kPairingRequest),
      service_name_(service_name),
      client_name_("") {
}

PairingRequestMessage::PairingRequestMessage(const std::string& service_name,
                                             const std::string& client_name)
    : PoloMessage(PoloMessage::kPairingRequest),
      service_name_(service_name),
      client_name_(client_name) {
}

std::string PairingRequestMessage::service_name() const {
  return service_name_;
}

std::string PairingRequestMessage::client_name() const {
  return client_name_;
}

bool PairingRequestMessage::has_client_name() const {
  return client_name_.length() > 0;
}

std::string PairingRequestMessage::ToString() const {
  std::ostringstream ss;
  ss << "[PairingRequestMessage service_name=" << service_name_
      << ", client_name=" << client_name_ << "]";
  return ss.str();
}

}  // namespace message
}  // namespace pairing
}  // namespace polo

File: external/google-tv-pairing-protocol/java/src/com/google/polo/pairing/message/PairingRequestMessage.java
/*
 * Copyright (C) 2009 Google Inc.  All rights reserved.
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *      http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

package com.google.polo.pairing.message;

/**
 * Object implementing the internal representation of the protocol message
 * 'PAIRING_REQUEST'.
 */
public class PairingRequestMessage extends PoloMessage {
  
  /**
   * The service name being requested for pairing.
   */
  private final String mServiceName;

  /**
   * Client name.
   */
  private final String mClientName;

  public PairingRequestMessage(String serviceName) {
    this(serviceName, null);
  }

  public PairingRequestMessage(String serviceName, String clientName) {
    super(PoloMessage.PoloMessageType.PAIRING_REQUEST);
    mServiceName = serviceName;
    mClientName = clientName;
  }

  public String getServiceName() {
    return mServiceName;
  }

  public String getClientName() {
    return mClientName;
  }

  public boolean hasClientName() {
    return mClientName != null;
  }

  @Override
  public String toString() {
    return new StringBuilder()
        .append("[")
        .append(getType())
        .append(" service_name=")
        .append(mServiceName)
        .append(", client_name=")
        .append(mClientName)
        .append("]").toString();
  }
  
  @Override
  public boolean equals(Object obj) {
    if (this == obj) {
      return true;
    }

    if (!(obj instanceof PairingRequestMessage)) {
      return false;
    }
    
    PairingRequestMessage other = (PairingRequestMessage) obj;
    
    if (mServiceName == null) {
      if (other.mServiceName != null) {
        return false;
      }
    } else if (!mServiceName.equals(other.mServiceName)) {
      return false;
    } else if (mClientName == null) {
      if (other.mClientName != null) {
        return false;
      }
    } else if (!mClientName.equals(other.mClientName)) {
      return false;
    }

    return true;
  }
  
}
